scripts/irods/test/test_iadmin.py

```diff
@@ -2132,6 +2132,11 @@ def test_moduser_type_rodsgroup(self):
                                    'STDERR', 'SYS_NOT_ALLOWED')
         self.assertEqual('rodsuser', lib.get_user_type(self.admin, self.username))
 
+    def test_downgrade_of_service_account_user_is_not_allowed__6127(self):
+        self.assertEqual('rodsadmin', lib.get_user_type(self.admin, 'rods'))
+        desired_error_msg = "Cannot downgrade another rodsadmin"
+        self.admin.assert_icommand(['iadmin', 'moduser', 'rods', 'type', 'rodsuser'], 'STDERR_SINGLELINE', desired_error_msg)
```

Comment:
Add another assertion that tests downgrade to `groupadmin` too.

That completes testing for all user types.